Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -65,13 +65,17 @@ def cmd_output(*args)
 
   def cmd(*args)
     output = cmd_output(*args)
-    assert_equal 0, $?.exitstatus
+    status = $?.exitstatus
+    puts "\n#{output}" if status != 0
+    assert_equal 0, status
     output
   end
 
   def cmd_fail(*args)
     output = cmd_output(*args)
-    assert_equal 1, $?.exitstatus
+    status = $?.exitstatus
+    $stderr.puts "\n#{output}" if status != 1
```

Comment:
But you don't even know where the output came from, as `cmd_output` is capturing both standard output and standard error. And in my opinion the question is more about where this output should go to as seen from the perspective of the test suite and less about where it originally came from. If you agree, this should be the same output stream in both cases, though I'm not sure whether `$stdout` or `$stderr` is more suitable.